staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -285,13 +390,35 @@ func logWatchChannelErr(err error) {
 // - get current objects if initialRev=0; set initialRev to current rev
 // - watch on given key and send events to process.
 func (wc *watchChan) startWatching(watchClosedCh chan struct{}) {
-	if wc.initialRev == 0 {
+	listObjects := wc.initialRev == 0
+	if wc.initialRev > 0 && !wc.initialEventsEndBookmarkSent {
+		currentStorageRV, err := storage.GetCurrentResourceVersionFromStorage(wc.ctx, wc.watcher.storage, wc.watcher.newListFunc, wc.watcher.resourcePrefix, wc.watcher.objectType)
```

Comment:
This seems to be called already in getStartWatchResourceVersion - can we somehow deduplicate it?

Reply:
I don't think so. I think these cases are mutually exclusive.